### PR TITLE
Feature/terraformv12

### PR DESCRIPTION
**Pull request by jeevanjoseph**
- Stack is migrated to TF12
- AD auto selection (for free tier shapes)
- Infra layout changed to remove SVC GW and NAT GW

### CODE REVIEW

**Comment by junior** [comment]
Please PR to Develop first

File: deploy/basic/terraform/schema.yaml [hunk 1]
```diff
@@ -30,15 +29,6 @@ variables:
     title: "Compartment"
     description: "The compartment in which to create compute instance(s)"
 
-  availability_domain:
```
Comment (junior) [inline deploy/basic/terraform/schema.yaml]: Removed? The current command automatically get the Always Free resources AD?

File: deploy/basic/terraform/main.tf [hunk 2]
```diff
@@ -62,7 +65,7 @@ locals  {
   
   instance_shape = "VM.Standard.E2.1.Micro"
 
-  availability_domain = 3
```
Comment (junior) [inline deploy/basic/terraform/main.tf]: What happen if the user has no limits on any AD? (e.g.: not on home region).

File: deploy/basic/terraform/main.tf [hunk 3]
```diff
@@ -1,35 +1,38 @@
...
-    source_id   = "${local.images[var.region]}"
+    source_id   = local.images[var.region]
   }
 
-  metadata {
-    ssh_authorized_keys = "${var.ssh_public_key}"
-    user_data           = "${base64encode(data.template_file.mushop.rendered)}"
-    db_name             = "${oci_database_autonomous_database.mushop_autonomous_database.db_name}"
-    atp_pw              = "${random_string.autonomous_database_wallet_password.result}"
+  metadata = {
```
Comment (peterj) [inline deploy/basic/terraform/main.tf]: is this `=` a typo or?

File: deploy/basic/terraform/scripts/node.sh [hunk 4]
```diff
@@ -52,8 +52,9 @@ MUSHOP_APP_URI=$(curl -L http://169.254.169.254/opc/v1/instance/metadata | jq -j
 WALLET_URI=$(curl -L http://169.254.169.254/opc/v1/instance/metadata | jq -j ".wallet_par")
 
 # get artifacts from object storage
-get_object /root/wallet.zip $${WALLET_URI}
+get_object /root/wallet.64 $${WALLET_URI}
 # Setup ATP wallet files
+base64 --decode /root/wallet.64 >  /root/wallet.zip
```
Comment (peterj) [inline deploy/basic/terraform/scripts/node.sh]: nit (sorry :)) you have an extra space after `>`

File: deploy/basic/terraform/schema.yaml (hunk 1 above)
Comment (jeevanjoseph) [reply deploy/basic/terraform/schema.yaml]: yes

File: deploy/basic/terraform/main.tf (hunk 2 above)
Comment (jeevanjoseph) [reply deploy/basic/terraform/main.tf]: deploy will fail

File: deploy/basic/terraform/main.tf (hunk 3 above)
Comment (jeevanjoseph) [reply deploy/basic/terraform/main.tf]: No - this became a map key recently.

File: deploy/basic/terraform/scripts/node.sh (hunk 4 above)
Comment (jeevanjoseph) [reply deploy/basic/terraform/scripts/node.sh]: Will fix

File: deploy/basic/terraform/main.tf (hunk 2 above)
Comment (jeevanjoseph) [reply deploy/basic/terraform/main.tf]: can trials even switch regions ?

Comment (junior) [reply deploy/basic/terraform/main.tf]: yes. Another use case is that the user are on the home region, but is already using the Always-free resources

Comment (jeevanjoseph) [reply deploy/basic/terraform/main.tf]: When resources are already used, they will hit service limits. When they don’t have resources, ( like different region) they will get an error about the AD being invalid.

**Pull request merged**